src/components/IconButton/IconButton.tsx

```diff
@@ -31,13 +31,20 @@ export interface IconButtonProps extends ButtonProps {
    * on hover, and accessible to screen readers.
    */
   label: string;
+
+  /**
+   * Optional additional references
+   */
+  additionalReferences?: {
+    iconRef?: React.RefObject<HTMLElement>;
+  };
```

Comment:
In my opinion, these additional refs increase the API surface too much. I can't think of a use case for them. If a valid use case arises in the future, I'd be happy to reconsider this approach, but I wouldn't add it yet.

As a rule of thumb, I think that only interactive elements should receive refs. In the case of the Button, it's the `<button />` itself, on an Input it's the `<input />` element. WDYT?

Reply:
Agreed